src/main/c/waf_jni.c

```diff
@@ -1111,6 +867,131 @@ static bool _cache_single_class_weak(JNIEnv *env,
+    }
+
+    return builder;
+}
+
+JNIEXPORT void JNICALL Java_com_datadog_ddwaf_WafBuilder_removeConfigNative(JNIEnv *env, jclass clazz,
+                                                                        jobject builder, jstring path) {
+
+    UNUSED(clazz);
+    UNUSED(env);
+    if (builder && path) {
+        ddwaf_builder_remove_config(_get_builder_checked(env, clazz, builder), JNI(GetStringUTFChars, path, NULL), JNI(GetStringLength, path));
+    }
+}
+
+JNIEXPORT jboolean JNICALL Java_com_datadog_ddwaf_WafBuilder_addOrUpdateConfigNative(JNIEnv *env, jclass clazz,
+jobject builder, jstring path, jobject configuration, jobject diagnostics) {
+    ddwaf_object ddwaf_diagnostics;
+    ddwaf_object_invalid(&ddwaf_diagnostics);
+        struct _limits limits = {
+                .max_depth = 20,
+                .max_elements = 1000000,
+                .max_string_size = 1000000,
+        };
+    ddwaf_object ddwaf_configuration =_convert_checked(env, configuration, &limits, 0);
+    if (JNI(ExceptionCheck)) {
+        return JNI_FALSE;
+    }
+    uint32_t path_length = JNI(GetStringLength, path);
+    if (JNI(ExceptionCheck)) {
+        return JNI_FALSE;
+    }
+    ddwaf_builder ddwaf_builder = _get_builder_checked(env, clazz, builder);
+    const char *path_ddwaf = JNI(GetStringUTFChars, path, NULL);
+    if (JNI(ExceptionCheck)) {
+        return JNI_FALSE;
+    }
+    ddwaf_builder_add_or_update_config(ddwaf_builder, path_ddwaf, path_length, &ddwaf_configuration,
+                                                                                        &ddwaf_diagnostics);
+
+    ddwaf_object_free(&ddwaf_configuration);
+    if (memcmp(&ddwaf_diagnostics, &(ddwaf_object) {0},
+                                    sizeof(ddwaf_diagnostics)) != 0) {
+            jobject jrsi = output_convert_diagnostics_checked(env, &ddwaf_diagnostics);
+
+            if (JNI(ExceptionCheck)) {
+                java_wrap_exc("Error converting rule info structure");
+                goto error;
+            }
+            JNI(SetObjectArrayElement, diagnostics, 0, jrsi);
+            if (JNI(ExceptionCheck)) {
+                java_wrap_exc("Error setting reference for WafDiagnostics");
+                goto error;
+            }
+            JNI(DeleteLocalRef, jrsi);
+            if (JNI(ExceptionCheck)) {
+                java_wrap_exc("Error deleting internal reference for WafDiagnostics");
+                goto error;
+            }
+        }
+    ddwaf_object_free(&ddwaf_diagnostics);
+    return JNI_TRUE;
+
+    error:
+        ddwaf_object_free(&ddwaf_diagnostics);
+        return JNI_FALSE;
+}
+
+JNIEXPORT jlong JNICALL Java_com_datadog_ddwaf_WafBuilder_initBuilder(JNIEnv *env, jclass clazz, jobject config) {
+    UNUSED(clazz);
+    ddwaf_config ddwaf_configuration;
+    _convert_ddwaf_config_checked(env, config, &ddwaf_configuration);
+    ddwaf_builder builder = ddwaf_builder_init(&ddwaf_configuration);
```

Comment:
So after this line, the dispose function should be called for ddwaf_configuration